ftdc/custom_format.go

```diff
@@ -32,35 +32,31 @@ type schema struct {
 // writeSchema writes down names for metrics in the form of a json array. All subsequent calls to
 // `writeDatum` will assume this "header" representation until the next call to `writeSchema`. A
 // full description of the file format is recorded in `doc.go`.
-func writeSchema(schema *schema, output io.Writer) {
+func writeSchema(schema *schema, output io.Writer) error {
```

Comment:
The following changes are me bubbling errors back up from failure to write data.